### PR TITLE
Guan dev：changed the name of Matrix_3x3 to Tensor2_3D, SymmMatrix_3x3 to SymmTensor2_3D.

### PR DESCRIPTION
**Pull request by ju-liu**
This update `simply` changed the name of a frequently used class, with the purpose of avoiding confusion.

(Symm)Tensor2_3D and (Symm)Tensor4_3D should only be used in material modeling.

### CODE REVIEW

**Review by Goldeninscriptions — COMMENTED** [review]
在model里有许多:
Tensor2_3D Cinv(C); Cinv.inverse();
或许可以直接用非成员函数：
Tensor2_3D Cinv = inverse( C );

File: include/Tensor2_3D.hpp [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #define TENSOR2_3D_HPP
 // ============================================================================
 // Tensor2_3D.hpp
-// This is a 2th-order tensor in 3D. The components are stored in a 1-D array:
```
Comment (ju-liu) [inline include/Tensor2_3D.hpp]: Good! 👍

Comment (ju-liu) [reply include/Tensor2_3D.hpp]: @Goldeninscriptions I have merged this with master. Please continue your review in master branch.

Comment (Goldeninscriptions) [reply include/Tensor2_3D.hpp]: OK

**Pull request merged**